src/main/java/unicash/logic/commands/GetBudgetCommand.java

```diff
@@ -64,22 +62,22 @@ public CommandResult execute(Model model) throws CommandException {
         var interval = budget.getInterval().interval;
         Function<LocalDateTime, Integer> filter = getIntervalFilter(interval);
         String intervalString = getIntervalString(interval);
-        requireAllNonNull(filter, intervalString);
+        assert filter != null;
+        assert intervalString != null;
 
-        var calculatedRemainder = model
+        double calculatedRemainder = model
                 .getFilteredTransactionList()
                 .stream()
+                .filter(t -> t.getType().type.equals(TransactionType.EXPENSE))
                 .filter(t -> filter.apply(t.getDateTime().getDateTime()).equals(filter.apply(from)))
-                .map(this::getAmountByType)
+                .map(t -> -1 * t.getAmount().amount)
```

Comment:
I believe @elhy1999 wrote a `getAmountAsDouble()` to circumvent this way of retrieving the raw value that we can use here instead
```suggestion
                .map(t -> -1 * t.getAmountAsDouble())
```

Reply:
![Screenshot 2023-11-02 at 20 35 06](https://github.com/AY2324S1-CS2103-T16-3/tp/assets/29497029/f6297ee7-22ba-4cc2-8e0b-6d691a2ebeb1)

Haha, it's a wrapper around `.amount`, so it's the same thing

Reply:
@woojiahao Yup! It was written because we were accessing amount directly too many times, better to keep it consistent across the codebase and use that method instead, but up to you though since it isn't a major issue

Reply:
Ahh I see, I think there's quite a few callsites where it's `amount` directly. For now I think we can keep it and refactor in 1.4